src/components/modal/modal.test.tsx

```diff
@@ -0,0 +1,44 @@
+import { render, fireEvent } from '@testing-library/react-native';
+import React from 'react';
+import { Text } from 'react-native';
+
+import Modal from './modal';
+
+describe('Modal Component', () => {
+  test('renders when visible', () => {
+    const { getByText } = render(
+      <Modal
+        isVisible={true}
+        onClose={() => {}}
+        header={<Text>Header</Text>}
+        body={<Text>Body</Text>}
+        footer={<Text>Footer</Text>}
+      />,
+    );
+    expect(getByText('Header')).toBeTruthy();
+    expect(getByText('Body')).toBeTruthy();
+    expect(getByText('Footer')).toBeTruthy();
+  });
+
+  test('does not render when not visible', () => {
+    const { queryByText } = render(
+      <Modal
```

Comment:
The usage should be:

```
<Modal>
  <Modal.Header></Modal.Header>
  <Modal.Body></Modal.Body>
  <Modal.Footer></Modal.Footer>
</Modal>
```

Header, Body, and Footer are not really props. They are children. In terms of OOPS, you should think about props like properties of an object.

`BaseWeb` provides a clean interface that can guide us on how to design interfaces of a component. You are welcome to use others. You are welcome to use Tailwind native css to design them.